javascript/code-challenges/challenges-11.test.js

```diff
@@ -0,0 +1,256 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named transformToLis that, given an object, returns an array of the key value pairs as html list items.
+
+For example:
+{
+  name: 'bob',
+  age: 32
+}
+
+Becomes:
+[
+<li>name: bob</li>,
+<li>age: 32</li>
+]
+------------------------------------------------------------------------------------------------ */
+
+function transformToLis(obj){
+  let answer = [];
+  if(obj.name){
+    let item1 = `<li>name: ${obj.name}</li>`;
+    let item2 = `<li>age: ${obj.age}</li>`;
+    answer.push(item1, item2);
+  }
+  return answer;
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named count that, given an integer and an array of arrays, uses either filter, map, or reduce to count the amount of times the integer is present in the array of arrays.
+
+Note: You might need to use the same method more than once.
+
+For example, count(5, [[1, 3, 5, 7, 9], [5, 5, 5], [1, 2, 3]]) returns 4.
+------------------------------------------------------------------------------------------------ */
+
+const count = (target, input) => {
+  let acc = 0;
+  for (let array of input){
+    for (let number of array){
+      if (number === target){
+        acc += 1;
+      }
+    }
+  }
+  return acc;
```

Comment:
Nice work on the challenges! 

This does pass the test of course, but the challenge wants you to use one of the higher order functions (filter, map, reduce, etc).

Here's a nudge/food for thought:
- you can use reduce() twice in this function, this is one way to tackle it
- The item we first iterate over at this time is the `input` arr (`const value = input.reduce((acc, curr) => {`)
- You then use a second reduce() (yes crazy, inside the first reduce) which takes in a result and a num as a callback
  `const countTargets = curr.reduce((result, num) => {`
- if the num === target, add it to results, then return result
- You always have to return accumultor (acc) in reduce , and for this you will need to return both acc and count
- The initial value in reduce would be 0 as we want to return an integer ( a count)

This may help a bit, but it just some food for thought for later if you want to come back and get some more practice.